test/src/commands/packages_test.dart

```diff
@@ -265,6 +266,64 @@ void main() {
           }).called(2);
         }),
       );
+
+      test(
+        'completes normally '
+        'when pubspec.yaml exists and directory is ignored (recursive)',
+        withRunner((commandRunner, logger, pubUpdater, printLogs) async {
+          final tempDirectory = Directory.systemTemp.createTempSync();
+          final directoryA = Directory(
+            path.join(tempDirectory.path, 'plugin_a'),
+          );
+          final directoryB = Directory(
+            path.join(tempDirectory.path, 'plugin_b'),
+          );
+          final pubspecA = File(
+            path.join(directoryA.path, 'example_a', 'pubspec.yaml'),
+          );
+          final pubspecB = File(
+            path.join(directoryB.path, 'example_b', 'pubspec.yaml'),
+          );
+          pubspecA
+            ..createSync(recursive: true)
+            ..writeAsStringSync(
+              '''
+          name: example_a
+          version: 0.1.0
+          
+          environment:
+            sdk: ">=2.12.0 <3.0.0"
+          ''',
+            );
+          pubspecB
+            ..createSync(recursive: true)
+            ..writeAsStringSync(
+              '''
+          name: example_b
+          version: 0.1.0
+          
+          environment:
+            sdk: ">=2.12.0 <3.0.0"
+          ''',
+            );
+
+          final result = await commandRunner.run(
+            [
+              'packages',
+              'get',
+              '--recursive',
+              '--ignore=plugin_b',
+              tempDirectory.path,
+            ],
+          );
+          expect(result, equals(ExitCode.success.code));
+          verify(() {
```

Comment:
Shouldn't we verify that we're running flutter packages get in the specific packages? The current verify isn't very specific and this test would likely pass even without the changes in the branch.